### ISSUE TITLE
Can't Add Homebridge to Google Home App 

### ISSUE THREAD

**Issue by sthety1** [homebridge-plugins/homebridge-gsh#7]
When in the Google Home app, when I try to add Homebridge, it shows connected and then an error stating 'Something went Wrong'

**To Reproduce:**

**Expected behavior:**

**Logs:**

```
Show the Homebridge logs here.
Remove any sensitive information.
```

**Homebridge Config:**

```json
Show your homebridge config.json here.
Remove any sensitive information, such as your homebridge-gsh / google-smarthome token.
```

**Screenshots:**

**Environment:**

- **Node.js Version**: 
- **NPM Version**: 
- **Homebridge Version**: 
- **Homebridge GSH Plugin Version**:
- **Homebridge Config UI X Plugin Version**:
- **Operating System**:

**Comment by NorthernMan54** [comment]
That’s odd, I just unlinked and relinked my account without issue

can you share a screenshot of the error?  I’m wondering where in the process it is erroring

**Comment by sthety1** [comment]
I'm using the same google account for HB and within Google home and this happens. Here is a recording. I've tried both Github and Google. 

https://github.com/user-attachments/assets/2d0b5cda-88f7-42a7-bf87-43497fd4f066

**Comment by sthety1** [comment]
Others have also said this on the discord channel as well.

**Comment by NorthernMan54** [comment]
At that point in the authentication / setup flow it is requesting the devices from your setup.  Is the homebridge-gsh plugin installed and linked with the same account ?  Any error logs from the plugin ?

The error you're seeing is the retrieval of devices failing for your account.

**Comment by sthety1** [comment]
Ok thanks. I might try with a different google account to see if I get the same issues to link to HB and the Google Home app.

**Comment by peterchabs** [comment]
I'm having the same problem, Homebridge console also reports not finding any Accessories. 

https://github.com/user-attachments/assets/f1c3defe-0185-4570-8f7e-e75565ee01f3

**Comment by NorthernMan54** [comment]
In homebridge what do the logs for the plugin show ?  It should show connecting to the service and # accessories shared

**Comment by sthety1** [comment]
![Screenshot 2024-12-18 at 1 07 49 PM](https://github.com/user-attachments/assets/66448528-77b3-485e-ba80-109c5a648603)
This is what is shows. It won't bring any devices since I can't connect on the google home. I have tried multiple google accounts and even GitHub and made sure the account I am using in home bridge is the same one in the google home app.

**Comment by NorthernMan54** [comment]
Can you go back further in the log, or restart the plugin so that the # of discovered accessories is displayed

```
[17/12/2024, 20:29:48] [Google Smart Home] Discovered 79 accessories
[17/12/2024, 20:30:03] [Google Smart Home] Sending Sync Request
```

Also is your internet connection stable ?  Those lost connection messages could be an issue with the plugin getting access to the cloud servers.

**Comment by peterchabs** [comment]
> In homebridge what do the logs for the plugin show ? It should show connecting to the service and # accessories shared

It says connected, but 0 accessories found.

[12/19/2024, 9:49:07 AM] [Google Smart Home] No more instances discovered, publishing services
[12/19/2024, 9:49:07 AM] [Google Smart Home] Loaded 0 accessories from Homebridge - pre filter
[12/19/2024, 9:49:07 AM] [Google Smart Home] Returned 0 accessories from Homebridge - post filter
[12/19/2024, 9:49:07 AM] [Google Smart Home] Discovered 0 accessories
[12/19/2024, 9:49:07 AM] [Google Smart Home] Monitoring 0 services for changes

[12/19/2024, 9:50:24 AM] [Google Smart Home] Received SYNC intent
[12/19/2024, 9:50:24 AM] [Google Smart Home] {
  "inputs": [
    {
      "intent": "action.devices.SYNC",
      "requestId": "17952288331438640880"
    }
  ],
  "requestId": "17952288331438640880"
}
[12/19/2024, 9:50:24 AM] [Google Smart Home] No supported devices found. See https://git.io/JfuHW
[12/19/2024, 9:50:26 AM] [Google Smart Home] Received SYNC intent
[12/19/2024, 9:50:26 AM] [Google Smart Home] No supported devices found. See https://git.io/JfuHW
[12/19/2024, 9:50:26 AM] [Google Smart Home] {
  "inputs": [
    {
      "intent": "action.devices.SYNC",
      "requestId": "15424523751066751156"
    }
  ],
  "requestId": "15424523751066751156"
}
[12/19/2024, 9:50:26 AM] [Google Smart Home] Received SYNC intent
[12/19/2024, 9:50:26 AM] [Google Smart Home] {
  "inputs": [
    {
      "intent": "action.devices.SYNC",
      "requestId": "15159484567186835292"
    }
  ],
  "requestId": "15159484567186835292"
}
[12/19/2024, 9:50:26 AM] [Google Smart Home] No supported devices found. See https://git.io/JfuHW
[12/19/2024, 9:50:27 AM] [Google Smart Home] Received SYNC intent
[12/19/2024, 9:50:27 AM] [Google Smart Home] No supported devices found. See https://git.io/JfuHW
[12/19/2024, 9:50:27 AM] [Google Smart Home] {
  "inputs": [
    {
      "intent": "action.devices.SYNC",
      "requestId": "3682975212494417584"
    }
  ],
  "requestId": "3682975212494417584"
}

**Comment by NorthernMan54** [comment]
@peterchabs This is why linking is failing, the plugin can't find any accessories.  What does the Homebridge UI show ?

Also this plugin takes homerbidge devices and exposes them to google.  You do have homebridge running, with plugins configured etc?

**Comment by peterchabs** [comment]
> @peterchabs This is why linking is failing, the plugin can't find any accessories. What does the Homebridge UI show ?
> 
> Also this plugin takes homerbidge devices and exposes them to google. You do have homebridge running, with plugins configured etc?

Yes, I do. I'm just trying to retrieve the single TP-Link accessory that is shown in the video I originally posted above.
<img width="1680" alt="Screenshot 2024-12-19 at 10 20 19 AM" src="https://github.com/user-attachments/assets/a37771f8-698b-417d-865c-ada763f941be" />
<img width="1680" alt="Screenshot 2024-12-19 at 10 21 13 AM" src="https://github.com/user-attachments/assets/61cbd64f-3e17-4fd8-8ce8-86310f00f186" />
<img width="1680" alt="Screenshot 2024-12-19 at 10 27 15 AM" src="https://github.com/user-attachments/assets/e8da82b7-ca5b-4a8c-a887-88a4baceddab" />

**Comment by NorthernMan54** [comment]
I'm kinda stumped here, as the plugin uses the same accessory search code as the UI, so it should be the same or similar.  Are you running child bridges ?  I'm wondering if it's a sequence issue during startup, and if your running child bridges you could restart the GSH Bridge to have it do discover after everythings running

**Comment by NorthernMan54** [comment]
Also could you enable DEBUG in the plugin config, and share the log from the first minute ( The discovery phase )
